### PR TITLE
Fix execution tests

### PR DESCRIPTION
**Pull request by camilbancioiu**
* Changed execution tests to do the correct asserts for setting `ElrondSyncExecAPIErrorShouldFailExecution` either true or false.
* Fixed `TestExecution_ExecuteOnSameContext_Simple`, which was broken.

### CODE REVIEW

File: arwen/contexts/runtime.go [hunk 1]
```diff
@@ -461,6 +461,10 @@ func (context *runtimeContext) ElrondAPIErrorShouldFailExecution() bool {
 	return true
 }
 
+func (context *runtimeContext) ElrondSyncExecAPIErrorShouldFailExecution() bool {
+	return false
```
Comment (sasurobert) [inline arwen/contexts/runtime.go:465]: this should not be false.

File: arwen/elrondapi/elrondei.go [hunk 2]
```diff
@@ -1291,18 +1291,18 @@ func executeOnSameContext(
 
 	sender := runtime.GetSCAddress()
 	destination, err := runtime.MemLoad(addressOffset, arwen.AddressLen)
-	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+	if arwen.WithFault(err, context, runtime.ElrondSyncExecAPIErrorShouldFailExecution()) {
```
Comment (sasurobert) [inline arwen/elrondapi/elrondei.go:1294]: no need for this

Comment (sasurobert) [reply arwen/elrondapi/elrondei.go:1294]: make it back

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: arwen/host/execution_test.go [hunk 3]
```diff
@@ -772,8 +797,16 @@ func TestExecution_ExecuteOnDestContext_Wrong(t *testing.T) {
 
 	vmOutput, err := host.RunSmartContractCall(input)
 	require.Nil(t, err)
-	expectedVMOutput := expectedVMOutput_DestCtx_WrongContractCalled(parentCode)
-	require.Equal(t, expectedVMOutput, vmOutput)
+	require.NotNil(t, vmOutput)
+
+	if host.Runtime().ElrondSyncExecAPIErrorShouldFailExecution() == false {
```
Comment (andrei-marinica) [inline arwen/host/execution_test.go:802]: weird having == false, maybe make it !host..... or invert the conditions

**Review by andrei-marinica — APPROVED** [review]

File: arwen/host/execution_test.go (hunk 3 above)
Comment (sasurobert) [reply arwen/host/execution_test.go:802]: this is only to check that the code was there.

**Pull request merged**